pkg/controller/plan/validation.go

```diff
@@ -191,6 +197,19 @@ func (r *Reconciler) validateOpenShiftVersion(plan *api.Plan) error {
 	return nil
 }
 
+func (r *Reconciler) ensureSecretForProvider(plan *api.Plan) error {
+	if plan.Referenced.Provider.Source != nil &&
+		plan.Referenced.Secret == nil &&
```

Comment:
unrelated to this PR but it's really not expressive enough to have a single secret on plan.Referenced since it's hard to say whether it refers to the source or the destination provider...